seL4-platforms/builds.py

```diff
@@ -552,6 +553,14 @@ def summarise_junit(file_path: str) -> Tuple[int, List[str]]:
                   "-q", junit_results, parsed_junit_results]
 
 
+def sim_script(success: str, timeout=1200):
+    return [
+        "expect", "-c",
+        'spawn ./simulate; set timeout %d; expect { "%s" {exit 0} timeout {exit 1} }' %
```

Comment:
Nitpicking: could break the expect tcl commands down into a list:
```suggestion
        "; ".join([
             "spawn ./simulate",
             f"set timeout {timeout}",
             'expect { "%s" {exit 0} timeout {exit 1} }' % success'])
```

Reply:
I don't think that's clearer, and it's dangerous if later somebody wants to add something the is not `;` terminated and doesn't spot what exactly the `join` uses.

The commands in a script are a list, because the command line invocation at the consuming end expects a list, not because a list was nicer to write.

Reply:
I can see you point. Maybe give https://pexpect.readthedocs.io/en/stable/install.html a try to make this nice then.

Reply:
Using `pexpect` doesn't fit into the framework, instead of executing a list of external commands it would now sometimes need to do a python thing in between. That sounds very messy to me.

I find the current version perfectly fine and readable, to be honest. The problem in the previous script wasn't that it wasn't readable, it was that whoever wrote the original version 10 years ago didn't read the manual for `expect` properly.

Reply:
I'm biassed there because we did create a python framework around running simulation on QEMU and other simulators. At some point expect was not powerful enough and we needed a more interactive way.

Reply:
For a script specifically for simulators that makes a lot of sense. When we want to go there, I'd put that into a separate program that is then called from this framework, though. I don't think it should be part of the general CI harness, because then you also can't run it yourself on the command line.

Reply:
(Currently we're not planning to add that kind of complexity to the simulation runs, though)